packages/core/src/services/datasets/destroy.ts

```diff
@@ -0,0 +1,31 @@
+import { eq } from 'drizzle-orm'
+
+import { Dataset } from '../../browser'
+import { database } from '../../client'
+import { Result, Transaction } from '../../lib'
+import { DiskWrapper } from '../../lib/disk'
+import { datasets } from '../../schema'
+
+export async function destroyDataset(
+  {
+    dataset,
+    disk,
+  }: {
+    dataset: Dataset
+    disk: DiskWrapper
+  },
+  db = database,
+) {
+  const deleteResult = await disk.delete(dataset.fileKey)
```

Comment:
This deletes the file in disk (Filesystem or S3 or other clouds). Do you think we should do in another way, or is blocking the request fine?

Reply:
fine

Reply:
Let me try in production with S3 enabled and we can move it to a job